sky/packages/sky/lib/sky_tool

```diff
@@ -199,16 +214,17 @@ class StartSky(object):
                               'The directory packages/sky_engine must also '
                               'exist to locate %s.' % (os.path.basename(__file__), APK_NAME))
                 return 2
-            if args.local_build:
-                apk_path = os.path.join(os.path.normpath(args.sky_src_path), args.android_debug_build_path, 'apks', APK_NAME)
-            else:
-                apk_path = os.path.join(APK_DIR, APK_NAME)
             if not os.path.exists(apk_path):
                 logging.error('"%s" does not exist.' % apk_path)
                 return 2
 
             cmd = [ADB_PATH, 'install', '-r', apk_path]
             subprocess.check_call(cmd)
+            # record the SHA1 of the APK we just pushed
+            with tempfile.NamedTemporaryFile() as fp:
+                fp.write(source_sha1)
+                fp.seek(0)
```

Comment:
I copied that code verbatim from shelldb. Your guess seems reasonable since Python buffers by default and the code worked in my testing :-)